common/uefi_scripts/startup.nsh

```diff
@@ -78,13 +101,19 @@ for %p in 0 1 2 3 4 5 6 7 8 9 A B C D E F then
 endfor
 :DoneDebug
 
-for %j in 0 1 2 3 4 5 6 7 8 9 A B C D E F then
-    if exist FS%j:\acs_tests\bsa\bsa.nsh then
-        FS%j:\acs_tests\bsa\bsa.nsh
-        goto Donebsa
-    endif
-endfor
-:Donebsa
+if %BsaEnableDisable% == "Enable" then 
+    for %j in 0 1 2 3 4 5 6 7 8 9 A B C D E F then
+        if exist FS%j:\acs_tests\bsa\bsa.nsh then
+            if %config_enabled_for_automation_run% == "Enable" then
+                FS%j:\acs_tests\bsa\bsa.nsh true
```

Comment:
In startup.nsh 
     
    check 
      if config for automation is true
                 call bsa.nsh  with true
     else
                call bsa.nsh  with false

and then in each test.nsh 
        if flag was true 
         run parser-efi -bsa
         if bsa run is enabled, run bsa as per cmd
        else
          dont run